src/proxy/topic/ReliableTopicListenerRunner.ts

```diff
@@ -59,45 +59,55 @@ export class ReliableTopicListenerRunner<E> {
-                        this.sequenceNumber++;
-                    }
-                    setImmediate(this.next.bind(this));
+                if (this.cancelled) {
+                    return;
                 }
-            })
-            .catch((e) => {
-                let message: string;
-                if (e instanceof StaleSequenceError) {
-                    this.ringbuffer.headSequence().then((seq: Long) => {
-                        const newSequence = seq.toNumber();
-
-                        message = 'Topic "' + this.proxy.getName() + '" ran into a stale sequence. ' +
-                            ' Jumping from old sequence ' + this.sequenceNumber + ' to new sequence ' + newSequence;
-                        this.logger.warn('ReliableTopicListenerRunner', message);
-
-                        this.sequenceNumber = newSequence;
-                        setImmediate(this.next.bind(this));
-                    });
 
-                    return;
+                const nextSeq = result.getNextSequenceToReadFrom().toNumber();
+                const lostCount = nextSeq - result.getReadCount() - this.sequenceNumber;
+                if (lostCount !== 0) {
+                    this.logger.trace('ReliableTopicListenerRunner', 'Listener of topic: ' + this.proxy.getName()
+                        + ' lost ' + lostCount + ' messages');
                 }
 
-                message = 'Listener of topic "' + this.proxy.getName() + '" caught an exception, terminating listener. ' + e;
-                this.logger.warn('ReliableTopicListenerRunner', message);
+                for (let i = 0; i < result.size(); i++) {
+                    const msg = new Message<E>();
+                    const item = result.get(i);
+                    msg.messageObject = this.serializationService.toObject(item.payload);
+                    msg.publisher = item.publisherAddress;
+                    msg.publishingTime = item.publishTime;
+                    process.nextTick(this.listener, msg);
+                }
 
-                this.proxy.removeMessageListener(this.listenerId);
+                this.sequenceNumber = nextSeq;
+                this.scheduleNext();
+            })
+            .catch((err) => {
+                if (this.handleInternalError(err)) {
+                    this.scheduleNext();
+                } else {
+                    this.logger.warn('ReliableTopicListenerRunner', 'Listener of topic: ' + this.proxy.getName()
+                        + ' caught an exception, terminating listener. ' + err);
+                    this.proxy.removeMessageListener(this.listenerId);
+                }
             });
     }
 
     public cancel(): void {
         this.cancelled = true;
     }
 
+    private scheduleNext(): void {
+        setImmediate(this.next.bind(this));
+    }
+
+    private handleInternalError(err: Error): boolean {
+        if (err instanceof InvocationTimeoutError) {
```

Comment:
There was an issue reported to the main repo. https://github.com/hazelcast/hazelcast/issues/17070

Maybe we could also handle `ClientNotActiveError` and log it in `trace` level instead of `warn`.

Reply:
Thanks for the heads-up. Changed the error logging in 9d6b40d274bcbaf29f9d05a40009bdf2c807736b